### PR TITLE
Reorganize and update language reference for attributes processed by the C# compiler, including nullable analysis

### LINKED ISSUE #16488: Missing Attributes
The attributes section is missing `DoesNotReturn` and `DoesNotReturnIf`: https://source.dot.net/#System.Private.CoreLib/NullableAttributes.cs,4cd15111d8d6eed4

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17290c4d-73f0-5ae1-e8ba-c039fe9fe9ca
* Version Independent ID: 59e0e3bb-b2e5-3ced-72d9-42e641b3897f
* Content: [Upgrade APIs for nullable reference types with attributes that define expectations for null values](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-attributes)
* Content Source: [docs/csharp/nullable-attributes.md](https://github.com/dotnet/docs/blob/master/docs/csharp/nullable-attributes.md)
* Product: **dotnet-csharp**
* Technology: **csharp-null-safety**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #17356: Split out strategies for approach nullability from the attributes themselves
I hit this page last night looking for some attributes to fix some nullable warnings I hit, and skipped it because the first bullet was "Choosing a nullable strategy" with the thinking that I'd already picked one so this page wasn't going to help me. I've been since pointed out this was the page that was going to help me. I think we should pull the attributes out separately from the approach because you are going to use them at different points in the lifecycle.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17290c4d-73f0-5ae1-e8ba-c039fe9fe9ca
* Version Independent ID: 59e0e3bb-b2e5-3ced-72d9-42e641b3897f
* Content: [Upgrade APIs for nullable reference types with attributes that define expectations for null values](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-attributes#conclusions)
* Content Source: [docs/csharp/nullable-attributes.md](https://github.com/dotnet/docs/blob/master/docs/csharp/nullable-attributes.md)
* Product: **dotnet-csharp**
* Technology: **csharp-null-safety**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #16488 
Fixes #17356 

This PR:

- Move global assembly attributes to language reference
- Move general reserved attributes to language reference
- Move Caller info into language reference
- Move nullable attributes content
- Add missing nullable attributes

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
Are the new files in this PR just renames that Git didn't recognize as such?  I took a cursory look assuming so -- if there were significant changes to the text beyond just renaming, let me know and I'll take a closer look. Otherwise, LGTM aside from the broken link warnings.

File: docs/csharp/language-reference/attributes/general.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+::::::code language="csharp" source="snippets/ConditionalExamples.cs" id="SnippetMultipleConditions" :::
+
+### Using Conditional with Attribute Classes
+
+The `Conditional` attribute can also be applied to an attribute class definition. In the following example, the custom attribute `Documentation` will only add information to the metadata if `DEBUG` is defined.
+
+::::::code language="csharp" source="snippets/ConditionalExamples.cs" id="SnippetConditionalConditionalAttribute" :::
+
+## Obsolete Attribute
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/general.md:36]:
```suggestion
## `Obsolete` attribute
```

File: docs/csharp/language-reference/attributes/general.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+In the following example the `Obsolete` attribute is applied to class `A` and to method `B.OldMethod`. Because the second argument of the attribute constructor applied to `B.OldMethod` is set to `true`, this method will cause a compiler error, whereas using class `A` will just produce a warning. Calling `B.NewMethod`, however, produces no warning or error. For example, when you use it with the previous definitions, the following code generates two warnings and one error:
+
+::::::code language="csharp" source="snippets/ObsoleteExample.cs" interactive="try-dotnet" :::
+
+The string provided as the first argument to the attribute constructor will be displayed as part of the warning or error. Two warnings for class `A` are generated: one for the declaration of the class reference, and one for the class constructor. The `Obsolete` attribute can be used without arguments, but including an explanation what to use instead is recommended.
+
+## AttributeUsage attribute
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/general.md:46]:
```suggestion
## `AttributeUsage` attribute
```

File: docs/csharp/language-reference/attributes/general.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+:::code language="csharp" source="snippets/NonInheritedAttribute.cs" id="SnippetNonInherited" :::
+
+In this case `NonInheritedAttribute` isn't applied to `DClass` via inheritance.
+
+## See also
+
+- <xref:System.Attribute>
+- <xref:System.Reflection>
+- [Attributes](../../../standard/attributes/index.md)
+- [Reflection (C#)](../../programming-guide/concepts/reflection.md)
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/general.md:98]:
```suggestion
- [Reflection](../../programming-guide/concepts/reflection.md)
```

File: docs/csharp/language-reference/attributes/global.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+title: "C# Reserved attributes: Global attributes"
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/global.md:2]:
```suggestion
title: "C# reserved attributes: Global attributes"
```

File: docs/csharp/language-reference/attributes/global.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Global attributes appear in the source code after any top level `using` directives and before any type, module, or namespace declarations. Global attributes can appear in multiple source files, but the files must be compiled in a single compilation pass. Visual Studio adds global attributes to the AssemblyInfo.cs file in .NET Framework projects. These attributes aren't added to .NET Core projects.
+
+Assembly attributes are values that provide information about an assembly. They fall into the following categories:
+
+- Assembly identity attributes
+- Informational attributes
+- Assembly manifest attributes
+
+## Assembly Identity Attributes
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/global.md:22]:
```suggestion
## Assembly identity attributes
```

File: docs/csharp/language-reference/attributes/global.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+The following table shows the identity attributes.
+
+|Attribute|Purpose|
+|---------------|-------------|
+|<xref:System.Reflection.AssemblyVersionAttribute>|Specifies the version of an assembly.|
+|<xref:System.Reflection.AssemblyCultureAttribute>|Specifies which culture the assembly supports.|
+|<xref:System.Reflection.AssemblyFlagsAttribute>|Specifies whether an assembly supports side-by-side execution on the same computer, in the same process, or in the same application domain.|
+
+## Informational Attributes
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/global.md:34]:
```suggestion
## Informational attributes
```

File: docs/csharp/language-reference/attributes/global.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+|---------------|-------------|
+|<xref:System.Reflection.AssemblyProductAttribute>|Specifies a product name for an assembly manifest.|
+|<xref:System.Reflection.AssemblyTrademarkAttribute>|Specifies a trademark for an assembly manifest.|
+|<xref:System.Reflection.AssemblyInformationalVersionAttribute>|Specifies an informational version for an assembly manifest.|
+|<xref:System.Reflection.AssemblyCompanyAttribute>|Specifies a company name for an assembly manifest.|
+|<xref:System.Reflection.AssemblyCopyrightAttribute>|Defines a custom attribute that specifies a copyright for an assembly manifest.|
+|<xref:System.Reflection.AssemblyFileVersionAttribute>|Sets a specific version number for the Win32 file version resource.|
+|<xref:System.CLSCompliantAttribute>|Indicates whether the assembly is compliant with the Common Language Specification (CLS).|
+
+## Assembly Manifest Attributes
```
Comment (pkulikov) [inline docs/csharp/language-reference/attributes/global.md:48]:
```suggestion
## Assembly manifest attributes
```

**Review by pkulikov — COMMENTED** [review]
Went through the style of titles

**Comment by BillWagner** [comment]
@tdykstra 

> Are the new files in this PR just renames that Git didn't recognize as such?

They were mostly renamed. The last section in the nullable-analysis.md file is new (`DoesNotReturn` and `DoesNotReturnIf`). Other than that, it was moving, then running acrolinx, improving the scores and fixing outdated styles.

I goofed by moving the files instead of using git mv. Let me know if you want to review more carefully. Otherwise, I'll :shipit: later today.

**Comment by tdykstra** [comment]
:shipit:

**Pull request merged**